kenya_compliance/kenya_compliance/overrides/server/stock_ledger_entry.py

```diff
@@ -425,11 +438,14 @@ def get_notes_docs_items_details(
 
 
 def get_warehouse_branch_id(warehouse_name: str) -> str | Literal[0]:
-    branch_id = frappe.db.get_value(
-        "Warehouse", {"name": warehouse_name}, ["custom_branch"], as_dict=True
-    )
+    try:
+        branch_id = frappe.db.get_value(
+            "Warehouse", {"name": warehouse_name}, ["custom_branch"], as_dict=True
+        )
 
-    if branch_id:
-        return branch_id.custom_branch
+        if branch_id:
+            return branch_id.custom_branch
+    except:
```

Comment:
always avoid catching general exceptions. Only catch errors/exception you will handle. If something should fail, let if fail.